### PR TITLE
Add definition for jest-plugin-context

### PR DESCRIPTION
**Pull request by aneilbaboo**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
   Output of `dts-gen -m jest-plugin-context -d types/jest-plugin-context` results in linter errors, so I modified the code using examples from other projects.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,5 +23,8 @@
   "devDependencies": {
     "dtslint": "github:Microsoft/dtslint#production",
     "types-publisher": "Microsoft/types-publisher#production"
+  },
+  "dependencies": {
+    "typescript": "^2.8.3"
   }
```
Comment (sheetalkamat) [inline package.json]: Please revert this unnecessary change

File: types/jest-plugin-context/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Type definitions for jest-plugin-context 2.9
+// Project: https://github.com/negativetwelve/jest-plugins/tree/master/packages/jest-plugin-context
+// Definitions by: Aneil Mallavarapu <https://github.com/aneilbaboo>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+
+/// <reference types="jest" />
+
+declare module 'jest-plugin-context' {
```
Comment (sheetalkamat) [inline types/jest-plugin-context/index.d.ts]: This isnt the correct way to declare this as module and global. See section https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-i-write-definitions-for-packages-that-can-be-used-globally-and-as-a-module and update accordingly.

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/jest-plugin-context/index.d.ts (hunk 2 above)
Comment (aneilbaboo) [reply types/jest-plugin-context/index.d.ts]: If I make those changes. I.e.:

```typescript
// Type definitions for jest-plugin-context 2.9
// Project: https://github.com/negativetwelve/jest-plugins/tree/master/packages/jest-plugin-context
// Definitions by: Aneil Mallavarapu <https://github.com/aneilbaboo>
// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
// TypeScript Version: 2.3

/// <reference types="jest" />

declare global {
  export const context: jest.Describe;
}

export default context;
```


I get the following errors:
```shell
$ npm run lint jest-plugin-context
Error: /Users/aneil/code/DefinitelyTyped/types/jest-plugin-context/index.d.ts:9:1
ERROR: 9:1   strict-export-declare-modifiers  Prefer 'export' to 'declare' in an external module. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md
ERROR: 13:1  strict-export-declare-modifiers  Prefer to explicitly write 'export' for an external module export. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md
```

Comment (aneilbaboo) [reply types/jest-plugin-context/index.d.ts]: Any thoughts on how / whether to address this?

**Comment by aneilbaboo** [comment]
@sheetalkamat - I made the changes you suggested, but these are producing errors in Travis. Any suggestions?
